drivers/power/adp1055/adp1055.c

```diff
@@ -0,0 +1,761 @@
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "adp1055.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+#include "no_os_crc8.h"
+#include "no_os_crc16.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Send command byte/word to ADP1055
+ * @param desc - ADP1055 device descriptor
+ * @param command - Value of the command
+ * @param return 0 in case of sucess, negative error code otherwise
+ */
+int adp1055_send_command(struct adp1055_desc *desc, uint16_t command)
+{
+	uint8_t data[2];
+	uint8_t command_val;
+
+	if (!desc)
+		return -EINVAL;
+
+	if (command > ADP1055_EXTENDED_COMMAND) {
+		data[0] = no_os_field_get(ADP1055_LSB_MASK, command);
+		data[1] = no_os_field_get(ADP1055_MSB_MASK, command);
+
+		return no_os_i2c_write(desc->i2c_desc, data, 2, 1);
+	}
+
+	command_val = no_os_field_get(ADP1055_LSB_MASK, command);
+
+	return no_os_i2c_write(desc->i2c_desc, &command_val, 1, 1);
+}
+
+/**
+ * @brief - Write data to ADP1055
+ * @param desc - ADP1055 device structure
+ * @param command - Command value
+ * @param data - Data value to write in ADP1055
+ * @return 0 in case of success, negative error code otherwise
+ */
+int adp1055_write(struct adp1055_desc *desc, uint16_t command, uint16_t data, uint8_t byte_num)
+{
+    	uint8_t val[4] = {0, 0, 0, 0};
+
+	if (!desc)
+		return -EINVAL;
+
+	if (command > ADP1055_EXTENDED_COMMAND) {
+		val[0] = no_os_field_get(ADP1055_MSB_MASK, command);
+		val[1] = no_os_field_get(ADP1055_LSB_MASK, command);
+		if (byte_num > 1) {
+			val[2] = no_os_field_get(ADP1055_LSB_MASK, data);
+			val[3] = no_os_field_get(ADP1055_MSB_MASK, data);
+		} else
+			val[2] = no_os_field_get(ADP1055_LSB_MASK, data);
+        
+        no_os_mdelay(52);
+		return no_os_i2c_write(desc->i2c_desc, val, byte_num + 2, 1);
+	} else {
+		val[0] = no_os_field_get(ADP1055_LSB_MASK, command);
+		if (byte_num > 1) {
+			val[1] = no_os_field_get(ADP1055_LSB_MASK, data);
+			val[2] = no_os_field_get(ADP1055_MSB_MASK, data);
+		} else
+			val[1] = no_os_field_get(ADP1055_LSB_MASK, data);
+
+        no_os_mdelay(52);
+		return no_os_i2c_write(desc->i2c_desc, val, byte_num + 1, 1);
+	}    
+}
+
+/**
+ * @brief Read data from ADP1055
+ * @param desc - ADP1055 device descriptor
+ * @param command - Command value
+ * @param data - Buffer with received data
+ * @return 0 if success, negative error otherwise
+ */
+int adp1055_read(struct adp1055_desc *desc, uint16_t command, uint8_t *data, uint8_t byte_num)
+{
+    	int ret;
+	uint8_t command_val[2] = {0, 0};
```

Comment:
You don't need to initialize this to 0, since the array elements get overwritten in the end. Also, if you have a case in which you need to initialize the array, you can do it as such `uint8_t arr[10] = {0}`